core/trino-main/src/main/java/io/trino/execution/executor/dedicated/SplitProcessor.java

```diff
@@ -73,35 +73,37 @@ public void run(SchedulerContext context)
         long previousScheduledNanos = 0;
         try (SetThreadName ignored = new SetThreadName("SplitRunner-%s-%s", taskId, splitId)) {
             while (!split.isFinished()) {
-                ListenableFuture<Void> blocked = split.processFor(SPLIT_RUN_QUANTA);
-                CpuTimer.CpuDuration elapsed = timer.elapsedTime();
-
-                long scheduledNanos = elapsed.getWall().roundTo(NANOSECONDS);
-                processSpan.setAttribute(TrinoAttributes.SPLIT_SCHEDULED_TIME_NANOS, scheduledNanos - previousScheduledNanos);
-                previousScheduledNanos = scheduledNanos;
-
-                long cpuNanos = elapsed.getCpu().roundTo(NANOSECONDS);
-                processSpan.setAttribute(TrinoAttributes.SPLIT_CPU_TIME_NANOS, cpuNanos - previousCpuNanos);
-                previousCpuNanos = cpuNanos;
-
-                if (!split.isFinished()) {
-                    if (blocked.isDone()) {
-                        processSpan.addEvent("yield");
-                        processSpan.end();
-                        if (!context.maybeYield()) {
-                            processSpan = null;
-                            return;
+                try (var ignored2 = processSpan.makeCurrent()) {
```

Comment:
Why does this fix the issue?

Reply:
`io.opentelemetry.api.trace.SpanBuilder#startSpan` doesn't propagate the context automatically in the current thread, it should be done manually (as mentioned in javadoc). 

From what I found, in Trino codebase it's done in several ways:
```
try (var ignored = span.makeCurrent()) {}
```
or 
```
try (var ignored = scopedSpan(span)) {}
```

BTW `ignored2` is not a best naming. But I believe when Trino migrates to Java 22 at a source level all these usage will be rewritten to just 
```
try (var _ = ...) {}
```